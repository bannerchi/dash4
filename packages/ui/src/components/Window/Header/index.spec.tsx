/// <reference types="@types/jest" />
import React from 'react';
import { render } from 'react-testing-library';
import { WindowHeader } from './index';

describe('WindowHeader', () => {
	test('exist', () => {
		expect(typeof WindowHeader).toBe('object');
	});
	test('render default', () => {
		const { container } = render(<WindowHeader title="title">Hello</WindowHeader>);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with subtitle', () => {
		const { container } = render(
			<WindowHeader title="title" subTitle="subtitle">
				Hello
			</WindowHeader>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render progressing', () => {
		const { container } = render(
			<WindowHeader progressing title="title" subTitle="subtitle">
				Hello
			</WindowHeader>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
