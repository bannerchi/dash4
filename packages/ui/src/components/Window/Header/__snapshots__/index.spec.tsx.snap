// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowHeader render default 1`] = `
.emotion-2 {
  position: relative;
  padding: 15px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-0 {
  width: 100%;
  white-space: normal;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2;
  margin-bottom: 0;
  overflow: hidden;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.emotion-1 {
  margin-left: 15px;
}

<div
  class=" emotion-2"
>
  <p
    class="emotion-0"
  >
    title
  </p>
  <div
    class="emotion-1"
  >
    Hello
  </div>
</div>
`;

exports[`WindowHeader render progressing 1`] = `
@keyframes animation-0 {
  0% {
    left: 0%;
    width: 0%;
  }

  50% {
    left: 40%;
    width: 50%;
  }

  90% {
    left: 90%;
    width: 10%;
  }

  91% {
    left: auto;
    right: 0%;
    width: 10%;
  }

  100% {
    left: auto;
    right: 0%;
    width: 0%;
  }
}

.emotion-4 {
  position: relative;
  padding: 15px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1 {
  width: 100%;
  white-space: normal;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2;
  margin-bottom: 0;
  overflow: hidden;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.emotion-2 {
  margin-left: 15px;
}

.emotion-0 {
  color: #6c757d;
  font-size: 80%;
  font-weight: 400;
}

.emotion-3 {
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  border-radius: 0;
  height: 1px;
  border: 0;
}

.emotion-3:after {
  content: ' ';
  position: absolute;
  top: 0;
  height: 1px;
  background-color: var(--primary);
  -webkit-animation: animation-0 1.5s infinite ease-in-out;
  animation: animation-0 1.5s infinite ease-in-out;
}

<div
  class=" emotion-4"
>
  <p
    class="emotion-1"
  >
    title
    <br />
    <small
      class="emotion-0"
    >
       
      subtitle
    </small>
  </p>
  <div
    class="emotion-2"
  >
    Hello
  </div>
  <div
    class="emotion-3"
  />
</div>
`;

exports[`WindowHeader render with subtitle 1`] = `
.emotion-3 {
  position: relative;
  padding: 15px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1 {
  width: 100%;
  white-space: normal;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2;
  margin-bottom: 0;
  overflow: hidden;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.emotion-2 {
  margin-left: 15px;
}

.emotion-0 {
  color: #6c757d;
  font-size: 80%;
  font-weight: 400;
}

<div
  class=" emotion-3"
>
  <p
    class="emotion-1"
  >
    title
    <br />
    <small
      class="emotion-0"
    >
       
      subtitle
    </small>
  </p>
  <div
    class="emotion-2"
  >
    Hello
  </div>
</div>
`;
